Rpackage/R/fitAndScore.R

```diff
@@ -0,0 +1,222 @@
+          alpha_path_score[[i]] <- rbind(alpha_path_score[[i]], data)
+        }
+      }
+      
+      colnames(alpha_path_score[[i]]) <- c("init alpha", " final alpha", "score")
+      alpha_path_score[[i]] <- tail(alpha_path_score[[i]], -1)
+      alpha_path_score[[i]] <- as.data.frame(alpha_path_score[[i]], row.names = "")
+    }
+    
+    # Prune the master tree based on the CV estimates
+    alphas <- NULL
+    alpha_cv_scores <- NULL
+    best_alpha <- -1.0
+    best_score <- attr(scorer, "worst")
+    best_tree <- NULL
+
+    for(i in 1 : length(master_alphas)){
+      if(i < length(master_alphas)){
+        geo_mean_alpha_k <- sqrt(master_alphas[[i]] * master_alphas[[i + 1]])
+      }
+      else{
+        geo_mean_alpha_k <- Inf
+      }
+
+      ### calc train score
+      prediction <- predict(master_pruned_trees[[i]], feature.mat)
+      train_score <- scorer(target.mat, prediction)
+      
+      ### calc cost of all leaves
+      ter_id <- nodeids(master_pruned_trees[[i]], terminal = TRUE)
+      n <- nodeapply(master_pruned_trees[[i]], ids = ter_id, info_node)
+      train_objective <- sum(matrix(unlist(n), nrow = length(n), byrow = T)[, 2])
+      
+      alpha_train_scores <- c(alpha_train_scores, train_score)
+      alpha_train_objective_values <- c(alpha_train_objective_values, train_objective)
+      
+      ### compute cv_score as mean of each fold scores
+      cv_score <- 0
+      flag <- 0
+      for(j in 1 : n_folds){
+        for(k in 1 : nrow(alpha_path_score[[j]])){
+          if((geo_mean_alpha_k < alpha_path_score[[j]][k, ]$` final alpha` || is.infinite(alpha_path_score[[j]][k, ]$` final alpha`)) && geo_mean_alpha_k >= alpha_path_score[[j]][k, ]$`init alpha`){
+            cv_score <- cv_score + alpha_path_score[[j]][k, ]$score
+            flag <- 1
+            break
+          }
+        }
+      }
+
+      assert_that(flag == 1, msg = "cv_score not updated")
+      cv_score <- cv_score/n_folds
+
+      # Log metrics for this alpha value
+      alphas <- c(alphas, geo_mean_alpha_k)
+      alpha_cv_scores <- c(alpha_cv_scores, cv_score)
+      
+      if(attr(scorer, "direction")(cv_score, best_score) == cv_score){
+        best_score <- cv_score
+        best_alpha <- geo_mean_alpha_k
+        best_tree <- master_pruned_trees[[i]]
+      }
+    }
+  }
+  else{
+    ### For each fold, build a decision tree
+    fold_test_scores <- NULL
+    for(i in 1 : n_folds){
+      prediction <- predict(fold_tree[[i]], feature.mat[fold_split_idx$test[i,],])
+      fold_test_scores <- c(fold_test_scores, scorer(target.mat, prediction))
+    }
+      
+    best_alpha <-  0.
+    best_score <- mean(fold_test_scores)
+    best_tree <- master_tree
+    alphas <- c(0.)
+    alpha_cv_scores <- c(best_score)
+    
+    ### calc train score
+    prediction <- predict(master_tree, feature.mat)
+    alpha_train_scores <- scorer(target.mat, prediction)
+    
+    ### calc cost of all leaves
+    if(learner == "mmif"){
```

Comment:
Add else if to explicitly capture mmit and fail otherwise. Will be more clear and robust to new algorithms.

Reply:
doen